### ISSUE TITLE
`Spanned` impl for `Span` type

### ISSUE THREAD

**Issue by cramertj** [alexcrichton/proc-macro2#185]
This would be quite handy to have, as a `Span` is trivially `Spanned`, but passing a `Span` into a function that takes `&dyn Spanned` or similar is quite hard. I'm not sure exactly where the impl would live, as I'm not sure whether it would make sense to have `Span` impl `ToTokens`, and there're coherence issues with having `Span` impl `Spanned` from `syn`, since `syn` can't know that `Span` won't start implementing `ToTokens` in the future.

**Comment by alexcrichton** [comment]
I think this might be a bug for the `syn` repository rather than this one? (`Spanned` is defined in `syn` I think)

**Comment by cramertj** [comment]
Well, that's sort of the question-- I'm not sure what repository (if any) have the ability to solve this problem. I was considering that perhaps the definition of `Span` could be moved into this library in order to solve the coherence problem, perhaps under a feature flag. cc @dtolnay who might have opinions here.

**Comment by dtolnay** [comment]
I haven't seen &dyn Spanned before. Could you explain a bit about the use case that requires a Spanned impl or &dyn Spanned rather than eagerly calling span() and keeping just the Span?

We could potentially move the Spanned trait into quote to resolve the coherence issue.

**Comment by cramertj** [comment]
You could certainly eagerly call `.span()`. I've been defining some small wrapper functions in terms of `&impl Spanned`, call `.span()`, and then call the inner function in order to get some minor ergonomics wins (since otherwise my code is littered with calls to `.span()` everywhere-- not a big deal, but kind of nice to avoid).

**Comment by dtolnay** [comment]
This is fixed in Syn 1.0.0.

```rust
use proc_macro2::Span;
use syn::spanned::Spanned;

fn f(_sp: &dyn Spanned) {}

fn main() {
    f(&Span::call_site());
}
```

**Comment by cramertj** [comment]
Wonderful, thanks!
